BrowserKit/Sources/ComponentLibrary/Buttons/SecondaryRoundedButton.swift

```diff
@@ -69,6 +69,7 @@ public class SecondaryRoundedButton: ResizableButton, ThemeApplicable {
         // use configuration.background.backgroundColor without any transformation
         updatedConfiguration.background.backgroundColorTransformer = nil
         updatedConfiguration.background.cornerRadius = UX.buttonCornerRadius
+        subviews.first?.layer.cornerRadius = UX.buttonCornerRadius
```

Comment:
Curious to clarify why the first layer gets set with square vs. corner radius as relying on the order of subviews can make the code more fragile to changes in the view hierarchy in future. 

Where is the square coming from, can we change and add corner radius there rather than first layer?

Reply:
That is a nice catch. Appreciate it. So, because the UIVisualEffectView is the problem, I am going to set the corner radius only for it.

<img width="340" alt="effectlayer" src="https://github.com/mozilla-mobile/firefox-ios/assets/61138287/d83bd7a7-f6c7-4341-a718-122e9746540d">